src/licensedcode/detection.py

```diff
@@ -1676,13 +1714,20 @@ def get_ambiguous_license_detections_by_type(unique_license_detections):
     ambi_license_detections = {}
 
     for detection in unique_license_detections:
+
         if not detection.license_expression:
             ambi_license_detections[DetectionCategory.LOW_QUALITY_MATCH_FRAGMENTS.value] = detection
 
+        elif can_ignore_ambiguous_detection(detection):
+            continue
+
         elif is_undetected_license_matches(license_matches=detection.matches):
             ambi_license_detections[DetectionCategory.UNDETECTED_LICENSE.value] = detection
 
-        elif has_correct_license_clue_matches(license_matches=detection.matches):
+        elif (
+            has_correct_license_clue_matches(license_matches=detection.matches) and
+            has_unknown_matches(license_matches=detection.matches)
```

Comment:
You may want to revisit the repeat call to has_unknown_matches below:
- https://github.com/aboutcode-org/scancode-toolkit/pull/4412/files#diff-5f6ccc1d8f9540cae715bddaf337246f4c69c48ca596b2086e05b18c541aef4aR1734